File: aliyun-java-best-practise/src/main/java/com/aliyun/bestpractise/disk/CreateDiskDemo.java
package com.aliyun.bestpractise.disk;

import com.aliyuncs.DefaultAcsClient;
import com.aliyuncs.IAcsClient;
import com.aliyuncs.ecs.model.v20140526.CreateDiskRequest;
import com.aliyuncs.ecs.model.v20140526.CreateDiskResponse;
import com.aliyuncs.exceptions.ClientException;
import com.aliyuncs.profile.DefaultProfile;

/**
 * 为ECS创建数据盘
 */
public class CreateDiskDemo {
    public static void main(String[] args) {
        DefaultProfile profile = DefaultProfile.getProfile(
                "RegionId",
                "accessKeyId",
                "accessKeySecret");
        IAcsClient client = new DefaultAcsClient(profile);
        CreateDiskRequest request = new CreateDiskRequest();
        //设置云盘所在区域
        request.setZoneId("cn-hangzhou-e");
        //设置云盘大小
        request.setSize(40);
        //设置云盘种类
        request.setDiskCategory("cloud_efficiency");
        CreateDiskResponse response;
        try {
            response =  client.getAcsResponse(request);
        } catch (ClientException e) {
            e.printStackTrace();
        }
    }
}

File: aliyun-net-best-practice/为ECS创建数据盘并挂载到服务器上/CreateDiskDemo.cs
using System;
using System.Collections.Generic;
using Aliyun.Acs.Core;
using Aliyun.Acs.Core.Exceptions;
using Aliyun.Acs.Core.Profile;
using Aliyun.Acs.Ecs.Model.V20140526;

namespace EcsDemo
{
    class CreateDiskDemo
    {
        public static void CreateDisk()
        {
            IClientProfile profile = DefaultProfile.GetProfile(
              "cn-hangzhou", "<accessKeyId>", "<accessSecret>"
            );
            DefaultAcsClient client = new DefaultAcsClient(profile);

            var request = new RunInstancesRequest();
            request.ImageId = "创建实例的模板ID";
            try {
                var response = client.GetAcsResponse(request);
                Console.WriteLine(System.Text.Encoding.Default.GetString(response.HttpResponse.Content));
            }
            catch (ServerException e)
            {
                Console.WriteLine(e);
            }
            catch (ClientException e)
            {
                Console.WriteLine(e);
            }
        }
    }
}